### DISCUSSION TITLE
Anyway to get Cloud Messaging analytics?

### DISCUSSION THREAD

**Discussion by paul-uz** [firebase/firebase-admin-node#1511] [Q&A]
Is there anyway to get analytics on messages sent, like what is visible in the Firebase Cloud Messaging dashboard?

**Comment by lahirumaramba** [comment]
Hi @paul-uz,
Did you mean to access [FCM aggregated delivery data](https://medium.com/firebase-developers/what-is-fcm-aggregated-delivery-data-d6d68396b83b) from the Admin SDK? We currently do not provide an Admin API for FCM aggregated delivery data, but you can access the APIs using the [Google API Client Libraries](https://developers.google.com/api-client-library). Please refer to the blog post linked above for more information.

cc: @chong-shao

**Comment by paul-uz** [comment]
@lahirumaramba yes, the data that is shown when you go to the Cloud Messaging dashboard.

Can I ask why is it not available in the Admin SDK? Seems like something that should be?

Whenever a message is sent, an ID is returned. What is that ID for exactly? Is it something that can be queried?
